### DISCUSSION TITLE
How to handle Errors during Actions

### DISCUSSION THREAD

**Discussion by globalflea** [serverlessworkflow/specification#491] [Q&A] [answered]
Would like to clarify what happens when an action (or more) fails amongst the set of actions to be done even after retrying?

With known/expected errors 
- Sequential action mode: do i just stop processing other actions and just return the error to the state?
- Parallel action mode: what to do with multiple errors that may arise due to parallelism?

**Comment by tsurdilo** [comment, marked as answer]
1. Sequential mode: If one invocation fails, and its retries are exhausted, the error is to be handled in the states onErrors definition. if error is not handled, workflow should fail. 

2. Parallel mode: actions are executed in parallel. If any fail and its retries are exhausted, the error is to be handled in the states onErrors definition. If error is not handled, workflow should fail.

You can read the spec doc about "autoretries" and "nonRetryableErrors" properties to see more info on retries.

Hope this helps.

**Reply by manick02** [reply]
A note to myself, I would need to resolve this as well -> https://github.com/serverlessworkflow/specification/issues/475 :)

**Comment by tsurdilo** [comment]
@cdavernas this question brings up the point again of adding the ability of actions groups in the future versions. This would give users the ability to retry a number of actions as a group, and not only individually as its currently possible, just fyi :)
